client-websocket/src/main/java/com/github/twitch4j/client/websocket/events/WebsocketStateEvent.java

```diff
@@ -0,0 +1,27 @@
+package com.github.twitch4j.client.websocket.events;
+
+import com.github.philippheuer.events4j.core.domain.Event;
+import com.github.twitch4j.client.websocket.domain.WebsocketConnectionState;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+/**
+ * Websocket State Event
+ * <p>
+ * Called when the websocket changes its connection state.
+ */
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class WebsocketStateEvent extends Event {
```

Comment:
Since events4j isn't meant to be referenced in [this module](https://github.com/twitch4j/twitch4j/blob/master/client-websocket/build.gradle.kts), perhaps the chat & pubsub modules should have their own module-specific event (& this would avoid the annoying cast from `AutoCloseable`)